src/Bicep.Core/Semantics/SymbolExtensions.cs

```diff
@@ -39,5 +38,36 @@ public static bool IsSecure(this ParameterSymbol parameterSymbol)
             }
             return false;
         }
+ 
+        public static string? GetDescription(this DeclaredSymbol symbol)
+        {
+            StatementSyntax? statement = symbol switch {
+                ParameterSymbol param => param.DeclaringParameter,
+                VariableSymbol var => var.DeclaringVariable,
+                ModuleSymbol mod => mod.DeclaringModule,
+                ResourceSymbol res => res.DeclaringResource,
+                OutputSymbol @out => @out.DeclaringOutput,
+                _ => null,
+            };
+
+            if (statement is null)
+            {
+                return null;
+            }
+
```

Comment:
I didn't explain myself well in the previous comment - I was imagining something more like the following (added to `SyntaxExtensions` rather than `SymbolExtensions` (note here I'm returning just the string, not markdown - as markdown formatting should be the responsibility of the language server handler, not here:

```csharp
public static string? TryGetDescription(this StatementSyntax statement)
{
    var descriptionSyntax = statement.Decorators
        .OfType<FunctionCallSyntax>()
        .Where(function => function.NameEquals("description"))
        .FirstOrDefault() as FunctionCallSyntax;

    if (descriptionSyntax is not null &&
        descriptionSyntax.Arguments.FirstOrDefault()?.Expression is StringSyntax stringSyntax)
    {
        return stringSyntax.TryGetLiteralValue();
    }

    return null;
}
```

In the hover handler, you're starting with a `StatementSyntax` - so it doesn't feel like there's a whole lot of point in going from syntax -> symbol -> syntax - may as well just use the syntax since you already have it.

Reply:
I moved this logic to `SyntaxExtensions.TryGetDecoratorSyntax`. 
In my opinion the "new" `syntax -> symbol -> syntax` is acceptable since the first syntax (SyntaxBase) -> symbol is happening anyways (in switch case), and the symbol -> syntax conversion is a no op and gives us a `StatementSyntax` (e.g. `module.DeclaringModule` -> `ModuleDeclarationSyntax`)

Reply:
I mean, it's obviously not the end of the world, but it's not optimal. My main issue that `StatementSyntax` is a more specific class than `SyntaxBase`. The `Symbol` class doesn't capture the fact that the syntax is actually a `StatementSyntax`, which forces you to include the switch statement to recover this information. Although it's *OK* for your particular usage with enough test coverage, other people may want to use this class, and capturing that the syntax must be a `StatementSyntax` provides additional safety.

IMO, as a general rule it's good to be as specific as possible with input data. `switch` on a type is not something we can avoid entirely, but we should avoid as much as possible. It's not too dissimilar to `cast` when you think about it, in that it makes it easier to accidentally introduce unsafe code.